tests/garage/tf/regressors/test_gaussian_mlp_regressor_with_model.py

```diff
@@ -0,0 +1,103 @@
+from nose2.tools.params import params
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.optimizers import PenaltyLbfgsOptimizer
+from garage.tf.regressors import GaussianMLPRegressorWithModel
+from tests.fixtures import TfGraphTestCase
+
+
+class TestGaussianMLPRegressorWithModel(TfGraphTestCase):
+    def test_fit(self):
+        gmr = GaussianMLPRegressorWithModel(input_shape=(1, ), output_dim=1)
+        data = np.linspace(-np.pi, np.pi, 1000)
+        obs = [{'observations': [[x]], 'returns': [np.sin(x)]} for x in data]
+
+        observations = np.concatenate([p['observations'] for p in obs])
+        returns = np.concatenate([p['returns'] for p in obs])
+        for _ in range(100):
+            gmr.fit(observations, returns.reshape((-1, 1)))
+
+        paths = {
+            'observations': [[-np.pi], [-np.pi / 2], [-np.pi / 4], [0],
+                             [np.pi / 4], [np.pi / 2], [np.pi]]
+        }
+
+        prediction = gmr.predict(paths['observations'])
+
+        expected = [[0], [-1], [-0.707], [0], [0.707], [1], [0]]
+        assert np.allclose(prediction, expected, rtol=0, atol=0.1)
+
+    def test_fit_custom(self):
+        gmr = GaussianMLPRegressorWithModel(
+            input_shape=(1, ),
+            output_dim=1,
+            subsample_factor=0.9,
+            normalize_inputs=False,
+            normalize_outputs=False)
+        data = np.linspace(-np.pi, np.pi, 1000)
+        obs = [{'observations': [[x]], 'returns': [np.sin(x)]} for x in data]
+
+        observations = np.concatenate([p['observations'] for p in obs])
+        returns = np.concatenate([p['returns'] for p in obs])
+        for _ in range(100):
+            gmr.fit(observations, returns.reshape((-1, 1)))
+
+        paths = {
+            'observations': [[-np.pi], [-np.pi / 2], [-np.pi / 4], [0],
+                             [np.pi / 4], [np.pi / 2], [np.pi]]
+        }
+
+        prediction = gmr.predict(paths['observations'])
+
+        expected = [[0], [-1], [-0.707], [0], [0.707], [1], [0]]
+        assert np.allclose(prediction, expected, rtol=0, atol=0.1)
+
+    def test_fit_without_trusted_region(self):
+        gmr = GaussianMLPRegressorWithModel(
+            input_shape=(1, ), output_dim=1, use_trust_region=False)
```

Comment:
What is the purpose of this test? To test if regressor works properly with `use_trust_region=False`?

Reply:
Yes

Reply:
I am confused. If PenaltyLbfgsOptimizer works well then this test will be correct. But PenaltyLbfgsOptimizer is never tested.